### PR TITLE
Make sure we only `import` directory

### PR DESCRIPTION
**Pull request by safareli**

### CODE REVIEW

File: test/src/Parse.purs [hunk 1]
```diff
@@ -157,6 +184,10 @@ testSuite1 = do
   """
 
   parseSucc """
+    import `foo/`; select * from `/test`
+  """
+
+  parseFail """
```
Comment (safareli) [inline test/src/Parse.purs:190]: *BREAKING*. Is there a way to not make it braking?

File: src/SqlSquared/Signature.purs [hunk 2]
```diff
@@ -761,16 +778,16 @@ genSqlF genLiteral n
...
   Gen.oneOf $ genImport :|
     [ genFunctionDecl n
     ]
 
-genSqlQueryF ∷ ∀ m. Gen.MonadGen m ⇒ CoalgebraM m SqlQueryF Int
+genSqlQueryF ∷ ∀ m. Gen.MonadGen m ⇒ MonadRec m ⇒ CoalgebraM m SqlQueryF Int
 genSqlQueryF n = Query <$> genDecls n <*> pure n
 
-genSqlModuleF ∷ ∀ m. Gen.MonadGen m ⇒ CoalgebraM m SqlModuleF Int
+genSqlModuleF ∷ ∀ m. Gen.MonadGen m ⇒ MonadRec m ⇒ CoalgebraM m SqlModuleF Int
```
Comment (safareli) [inline src/SqlSquared/Signature.purs:775]: *BREAKING*. As new constraint is introduced (that's needed for generating of path-es)

File: src/SqlSquared/Signature.purs [hunk 3]
```diff
@@ -138,8 +142,20 @@ data SqlF literal a
   | Select (SelectR a)
   | Parens a
 
+type AnyDirPath = E.Either (Pt.AbsDir Pt.Unsandboxed) (Pt.RelDir Pt.Unsandboxed)
+
+printAnyDirPath :: AnyDirPath -> String
+printAnyDirPath = E.either Pt.unsafePrintPath Pt.unsafePrintPath
+
+parseAnyDirPath :: forall m. Applicative m => (forall a. String -> m a) -> String -> m AnyDirPath
```
Comment (cryogenian) [inline src/SqlSquared/Signature.purs]: Can't we move `a` to the left?

File: src/SqlSquared/Signature.purs [hunk 4]
```diff
@@ -138,8 +142,20 @@ data SqlF literal a
   | Select (SelectR a)
   | Parens a
 
+type AnyDirPath = E.Either (Pt.AbsDir Pt.Unsandboxed) (Pt.RelDir Pt.Unsandboxed)
```
Comment (cryogenian) [inline src/SqlSquared/Signature.purs]: Might be worthing to introduce the same type and print/parse for filepath. It's used in table relation.

File: src/SqlSquared/Signature.purs [hunk 5]
```diff
@@ -878,16 +895,19 @@ genFunctionDecl n = do
   args ← L.foldM foldFn L.Nil $ L.range 0 len
   pure $ FunctionDecl { ident, args, body: n - 1 }
 
-genImport ∷ ∀ m a. Gen.MonadGen m ⇒ m (SqlDeclF a)
-genImport = Import <$> genIdent
+genImport ∷ ∀ m a. Gen.MonadGen m ⇒ MonadRec m ⇒ m (SqlDeclF a)
+genImport = map Import
+  $ Gen.oneOf
+  $ (Pt.unsandbox >>> E.Left <$> PtGen.genAbsDirPath)
```
Comment (cryogenian) [inline src/SqlSquared/Signature.purs]: Flipped flow. `Pt.unsandbox >>> E.Left` left-to-right, `<$>` right-to-left. It's hard to follow, would you mind switching to `E.Left <<< Pt.unsandbox <$> PtGen.genAbsDirPath` or `Pt.genAbsDirPath <#> Pt.unsandbox >>> E.Left`?

**Review by cryogenian — CHANGES_REQUESTED** [review]
Please, change `a >>> b <$> c` to `b <<< a <$> c` :pray: 

Also two small things to consider
+ wrapped `forall a.` -- I might be wrong but the quantifier could be moved to the left
+ It would be cool if we had `print/parse` for table relation

**Comment by cryogenian** [comment]
Actually ignore my comment about `forall a.` parens :)

File: src/SqlSquared/Signature.purs (hunk 3 above)
Comment (cryogenian) [reply src/SqlSquared/Signature.purs]: We should be able to do this
```
forall m. (forall a. String -> a) -> any ~ 
forall m. (exists a. (String -> a) -> any) ~ 
forall m a. (String -> a) -> any
```

File: src/SqlSquared/Signature.purs [hunk 6]
```diff
@@ -138,8 +142,20 @@ data SqlF literal a
...
+
+printAnyDirPath :: AnyDirPath -> String
+printAnyDirPath = E.either Pt.unsafePrintPath Pt.unsafePrintPath
+
+parseAnyDirPath :: forall m. Applicative m => (forall a. String -> m a) -> String -> m AnyDirPath
+parseAnyDirPath fail = Pt.parsePath
+  (pure ∘ E.Right)
+  (pure ∘ E.Left)
+  (const $ fail "incorrect directory path")
+  (const $ fail "incorrect directory path")
```
Comment (garyb) [inline src/SqlSquared/Signature.purs]: Nitpick: I think "incorrect directory path" is an easy error for users to misinterpret. Perhaps "Expected a directory path"?

Comment (safareli) [reply src/SqlSquared/Signature.purs]: Absolutely!

**Comment by safareli** [comment]
Made requested changes.

**Comment by cryogenian** [comment]
Awesome! Thank you very much!

**Review by cryogenian — APPROVED** [review]

**Comment by cryogenian** [comment]
Besides broken travis of course :P

**Comment by safareli** [comment]
So the braking part of the change is fine? i.e. queries which were fine will be rejected after upgrade.

File: src/SqlSquared/Signature.purs (hunk 3 above)
Comment (safareli) [reply src/SqlSquared/Signature.purs]: I moved a to top forall but it fails with:
```
[1/1 TypesDoNotUnify] src/SqlSquared/Path.purs:41:4

  41    (pure ∘ E.Right)
         ^^^^^^^^^^^^^^
  
  Could not match type
  
    Either t4 (Path Rel File Unsandboxed)
  
  with type
  
    a1
  
  while trying to match type t2 t3
    with type m0 a1
  while checking that expression (compose pure) Right
    has type Path Rel File Unsandboxed -> m0 a1
  in value declaration parseAnyFilePath
  
  where a1 is a rigid type variable
        m0 is a rigid type variable
        t3 is an unknown type
        t2 is an unknown type
        t4 is an unknown type
```
so reverting to inner forall

**Comment by safareli** [comment]
This is passing now.

What should be a new version? (`v0.9.1` is current one )

**Comment by cryogenian** [comment]
Yeah, breaking incorrect queries is absolutely fine, they didn't work in quasar anyways. 
I'd make it `0.10.0`
